### DISCUSSION TITLE
Script Components insert multiple script tags.

### DISCUSSION THREAD

**Discussion by hoo00nn** [vercel/next.js#37713] [Help] [answered]
Hi, I'm using Next.js well, but I've encountered some problems.

When you add a script tag using next/script, multiple identical scripts are inserted.

Are Script tags affected by rendering?

e.g

<img width="554" alt="image" src="https://user-images.githubusercontent.com/52775389/173812022-db3217f5-27ef-4697-b4ca-2b42e504448a.png">

 
```
const HeadElement: React.FC = () => {
  return (
    <>
      <Head>
          <title>blahblah</title>
      </Head>
      <Script strategy="lazyOnload" {...restProps}>
           {`console.log('hello, world')`}
       </Script>
  );
};
```

help me.

**Comment by icyJoseph** [comment, marked as answer]
> The id property is required for inline scripts in order for Next.js to track and optimize the script.

Place an `id` on the script. For more information please refer to the docs, https://nextjs.org/docs/basic-features/script#inline-scripts.

**Reply by hoo00nn** [reply]
@icyJoseph  Thank you for your answer. I have one more question.

When I used script component, I checked that it is always inserted in body tag.

Can it be inserted within the Head Tag?

We have confirmed that if you use a simple script tag rather than a script component within the Head component provided by next/head, it will be added within the Head tag. Is it okay to use it like this when I try to add it within the head tag?
